sdk/core/test/unit/services/BloomVaultStorageService.test.ts

```diff
@@ -56,16 +61,77 @@ describe('BloomVaultStorageService', () => {
         { id: 1, cyphertext: JSON.stringify({ ...signedCredential, type: ['type1'] }) },
       ])
     nock(bloomVaultUrl, { reqheaders }).get('/data/100/199').reply(200, [])
+    nock(VAULT_MIGRATION_SERVICE_URL, { reqheaders }).get('/migration/started').reply(200, 'false')
+    const service = createBloomStorageService()
+    const credentials = await service.searchCredentials(region)
+
+    expect(stubStatus.notCalled).to.be.true
+    expect(stubMigrationProcess.notCalled).to.be.true
+    expect(credentials).to.length(2)
+    expect(credentials[0].id).to.eql(signedCredential.id)
+  })
+
+  it('should return empty array if migration done', async () => {
+    sinon.stub(DidAuthService.prototype, 'pullDidAuthRequestToken').returns(Promise.resolve('requestToken'))
+    sinon.stub(DidAuthService.prototype, 'createDidAuthResponseToken').returns(Promise.resolve('responseToken'))
+    sinon
+      .stub(BloomVaultStorageService.prototype, 'didEthr')
+      .returns('did:ethr:0x042f98f56ad0ca5d5fecc26e3930df37cd5a5d8a')
+
+    await authorizeVault()
 
+    nock(VAULT_MIGRATION_SERVICE_URL, { reqheaders }).get('/migration/started').reply(200, 'true')
     const service = createBloomStorageService()
+    nock(VAULT_MIGRATION_SERVICE_URL, { reqheaders }).get(`/migration/done/${service.didEthr}`).reply(200, 'true')
     const credentials = await service.searchCredentials(region)
+
+    expect(credentials).to.length(0)
+  })
+
+  it('should call `MigrationHelper.runMigration` method only once if migration done endpoint return false', async () => {
+    sinon.stub(DidAuthService.prototype, 'pullDidAuthRequestToken').returns(Promise.resolve('requestToken'))
+    sinon.stub(DidAuthService.prototype, 'createDidAuthResponseToken').returns(Promise.resolve('responseToken'))
+
+    await authorizeVault()
+
+    nock(bloomVaultUrl, { reqheaders })
+      .get('/data/0/99')
+      .reply(200, [
+        { id: 0, cyphertext: JSON.stringify(signedCredential) },
+        { id: 1, cyphertext: JSON.stringify({ ...signedCredential, type: ['type1'] }) },
+      ])
+    nock(bloomVaultUrl, { reqheaders }).get('/data/100/199').reply(200, [])
+    const stub = sinon.stub(MigrationHelper.prototype, 'runMigration').resolves()
+    nock(VAULT_MIGRATION_SERVICE_URL, { reqheaders }).get('/migration/started').reply(200, 'true')
+    const service = createBloomStorageService()
+    nock(VAULT_MIGRATION_SERVICE_URL, { reqheaders }).get(`/migration/done/${service.didEthr}`).reply(200, 'false')
```

Comment:
probably here will be good to mock http call to POST `/migration/credentials` instead of stubbing it